tracing/tracing.go

```diff
@@ -6,51 +6,185 @@ package tracing
+		cfg.samplerType = e
+	}
+
+	if e := os.Getenv(envSamplerParam); e != "" {
+		if value, err := strconv.ParseFloat(e, 64); err == nil {
+			cfg.samplerParam = value
+		} else {
+			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envSamplerParam, e)
+		}
+	}
+
+	if e := os.Getenv(envSamplerRemoteURL); e != "" {
+		cfg.samplerRemoteURL = e
+	} else if e := os.Getenv(envJaegerAgentHost); e != "" {
+		// Fallback if we know the agent host - try the sampling endpoint there
+		cfg.samplerRemoteURL = fmt.Sprintf("http://%s:%d/sampling", e, DefaultSamplingServerPort)
+	}
+
+	cfg.customAttributes, err = parseAttributes(os.Getenv("JAEGER_TAGS"))
 	if err != nil {
-		return nil, errors.Wrap(err, "could not initialize jaeger tracer")
+		return nil, errors.Wrap(err, "could not parse JAEGER_TAGS")
 	}
-	return closer, nil
+	return &cfg, nil
 }
 
-// NewFromEnv is a convenience function to allow tracing configuration
-// via environment variables
-//
-// Tracing will be enabled if one (or more) of the following environment variables is used to configure trace reporting:
-// - JAEGER_AGENT_HOST
-// - JAEGER_SAMPLER_MANAGER_HOST_PORT
-func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, error) {
-	cfg, err := jaegercfg.FromEnv()
+func (cfg Config) initJaegerTracerProvider(serviceName string) (*tracesdk.TracerProvider, error) {
+	// Read environment variables to configure Jaeger
+	var ep jaegerotel.EndpointOption
+	// Create the jaeger exporter: address can be either agent address (host:port) or collector URL
+	if cfg.host != "" {
+		ep = jaegerotel.WithAgentEndpoint(jaegerotel.WithAgentHost(cfg.host), jaegerotel.WithAgentPort(strconv.Itoa(cfg.port)))
+	} else {
+		ep = jaegerotel.WithCollectorEndpoint(jaegerotel.WithEndpoint(cfg.collectorURL))
+	}
+	exp, err := jaegerotel.New(ep)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := NewResource(serviceName, cfg.customAttributes)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
+		return nil, err
 	}
 
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
-		return nil, ErrBlankTraceConfiguration
+	// Configure sampling strategy
+	sampler := tracesdk.AlwaysSample()
+	if cfg.samplerType == "const" || cfg.samplerType == "probabilistic" {
+		sampler = tracesdk.TraceIDRatioBased(cfg.samplerParam)
+	} else if cfg.samplerType == "remote" {
+		sampler = jaegerremote.New(serviceName, jaegerremote.WithSamplingServerURL(cfg.samplerRemoteURL),
+			jaegerremote.WithInitialSampler(tracesdk.TraceIDRatioBased(cfg.samplerParam)))
+	} else if cfg.samplerType != "" {
+		return nil, errors.Errorf("unknown sampler type %q", cfg.samplerType)
 	}
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithBatcher(exp),
+		tracesdk.WithResource(res),
+		tracesdk.WithSampler(sampler),
+	)
+
+	otel.SetTracerProvider(tp)
+	propagators := []propagation.TextMapPropagator{}
```

Comment:
nitpick: this can be preallocated or you can add all the propagators in the slice declaration 

```go
	propagators := []propagation.TextMapPropagator{
    propagation.TraceContext{}, propagation.Baggage{},
		jaegerpropagator.Jaeger{},
	}
```